ygen/codegen.go

```diff
@@ -409,13 +408,14 @@ func (cg *YANGCodeGenerator) GenerateGoCode(yangFiles, includePaths []string) (*
 	var enumTypeMapCode string
 	if cg.Config.GenerateJSONSchema {
 		var err error
-		if rawSchema, err = serialiseStructDefinitions(goStructs, cg.Config.GenerateFakeRoot, cg.Config.FakeRootName, cg.Config.CompressOCPaths); err != nil {
-			codegenErr.Errors = append(codegenErr.Errors, err)
+		rawSchema, err = buildJSONTree(mdef.modules, cg.state.uniqueDirectoryNames, mdef.directoryEntries["/"])
+		if err != nil {
+			codegenErr.Errors = append(codegenErr.Errors, fmt.Errorf("error marshalling JSON schema: %v", err))
 		}
 
-		if rawSchema != nil {
+		if len(rawSchema) > 0 {
```

Comment:
It's better to fix the method that returns this to return nil if the value has len 0, particularly since the value is used to make decisions.

Reply:
Done.